notebooks/ch14/bias_modeling.md

```diff
@@ -215,14 +254,20 @@ plt.title('Irreducible error');
 ```
 
 
-![png](bias_modeling_files/bias_modeling_20_0.png)
+![png](bias_modeling_files/bias_modeling_22_0.png)
 
 
 ## Bias-Variance In Practice
 
-In practice, we do not know the population data generation process and thus are unable to precisely determine a model's bias, variance, or irreducible error. Instead, we use our observed dataset as an approximation to the population. As we have seen, however, achieving a low training error does not necessarily mean that our model will have a low test error as well. Fundamentally, this occurs because training error reflects the bias of our model but not the variance; the test error reflects both.
+In an ideal world, we would minimize the expected prediction error for our model over all input-output points in the population.  However, in practice, we do not know the population data generation process and thus are unable to precisely determine a model's bias, variance, or irreducible error. Instead, we use our observed dataset as an approximation to the population. 
+
+As we have seen, however, achieving a low training error does not necessarily mean that our model will have a low test error as well. It is easy to obtain a model with extremely low bias and therefore low training error by fitting a curve that passes through every training observation. However, this model will have high variance, which generally leads to high test error. Conversely, you can obtain a model with very low variance by fitting a horizontal line to the data, but this leads to high bias. Fundamentally, this occurs because training error reflects the bias of our model but not the variance; the test error reflects both. In order to minimize test error, our model needs to simultaneously achieve low bias and low variance. To account for this, we need a way to simulate test error without using the test set. This is generally done using cross validation.
```

Comment:
nit: A few too many commas:

> However, this model will have high variance, which generally leads to high test error.

Should be 

> However, this model will have high variance which typically leads to high test error.

(Let's use the term "generally" very carefully since it has a specific meaning in statistics.)

------ 

> Conversely, you can obtain a model with very low variance by fitting a horizontal line to the data, but this leads to high bias.

Should be

> Conversely, a model that predicts a constant has low variance but high bias.

(In higher dimensions, predicting a constant isn't really a "horizontal line" anymore.)

Reply:
fixed